[Bug]: customConditions are not reset in `fixupCompilerOptionsForModuleKind`

### Version

29.2.5

### Steps to reproduce

use a tsconfig with

```
        "module": "preserve",
        "moduleResolution": "bundler",
        "customConditions": [
            "node",
        ],
```

### Expected behavior

ts-jest transpiles .ts files

### Actual behavior

```
  ● Test suite failed to run

    error TS5098: Option 'customConditions' can only be used when 'moduleResolution' is set to 'node16', 'nodenext', or 'bundler'.
```

### Debug log

-

### Additional context

https://github.com/kulshekhar/ts-jest/blob/main/src/legacy/compiler/ts-compiler.ts#L155 sets `moduleResolution: "node"` but it doesn't reset customConditions which are not supported with the new moduleResolution value. Adding `customConditions: undefined` works well.


### Environment

```shell
System:
    OS: Windows 10 10.0.26100
    CPU: (16) x64 Intel(R) Core(TM) i7
  Binaries:
    Node: 22.11.0 - ~\AppData\Local\nvs\default\node.EXE
    Yarn: 4.1.1 - ~\AppData\Local\nvs\default\yarn.CMD
    npm: 10.9.0 - ~\AppData\Local\nvs\default\npm.CMD
  npmPackages:
    jest: 29.7.0 => 29.7.0
```

--- a/src/legacy/compiler/ts-compiler.spec.ts
+++ b/src/legacy/compiler/ts-compiler.spec.ts
@@ -41,7 +41,9 @@ describe('TsCompiler', () => {
       const compiler = makeCompiler({
         tsJestConfig: {
           ...baseTsJestConfig,
-          isolatedModules: true,
+          tsconfig: {
+            isolatedModules: true,
+          },
         },
       })
 
@@ -111,7 +113,15 @@ describe('TsCompiler', () => {
         },
       ])('should transpile code with config %p', ({ useESM, babelConfig, supportsStaticESM }) => {
         const compiler = makeCompiler({
-          tsJestConfig: { ...baseTsJestConfig, isolatedModules: true, useESM, babelConfig },
+          tsJestConfig: {
+            ...baseTsJestConfig,
+            useESM,
+            babelConfig,
+            tsconfig: {
+              isolatedModules: true,
+              customConditions: ['my-condition'],
+            },
+          },
         })
         const transformersStub = {
           before: [],
@@ -138,12 +148,19 @@ describe('TsCompiler', () => {
           module: usedCompilerOptions.module,
           esModuleInterop: usedCompilerOptions.esModuleInterop,
           allowSyntheticDefaultImports: usedCompilerOptions.allowSyntheticDefaultImports,
+          customConditions: usedCompilerOptions.customConditions,
         }).toMatchSnapshot()
       })
 
       test.each([true, false])('should report diagnostics if shouldReportDiagnostics is %p', (shouldReport) => {
         const compiler = makeCompiler({
-          tsJestConfig: { ...baseTsJestConfig, isolatedModules: true, useESM: false },
+          tsJestConfig: {
+            ...baseTsJestConfig,
+            useESM: false,
+            tsconfig: {
+              isolatedModules: true,
+            },
+          },
         })
         compiler.configSet.raiseDiagnostics = jest.fn()
         compiler.configSet.shouldReportDiagnostics = jest.fn().mockReturnValue(shouldReport)
@@ -227,6 +244,7 @@ describe('TsCompiler', () => {
               tsconfig: {
                 module: moduleValue as unknown as RawCompilerOptions['module'],
                 esModuleInterop: false,
+                customConditions: ['my-condition'],
               },
             },
           })
@@ -253,6 +271,7 @@ describe('TsCompiler', () => {
           expect(usedCompilerOptions.module).toBe(expectedModule)
           expect(usedCompilerOptions.esModuleInterop).toBe(expectedEsModuleInterop)
           expect(usedCompilerOptions.moduleResolution).toBe(ts.ModuleResolutionKind.Node10)
+          expect(usedCompilerOptions.customConditions).toBeUndefined()
           expect(output).toEqual({
             code: updateOutput(jsOutput, fileName, sourceMap),
             diagnostics: [],
@@ -355,7 +374,13 @@ describe('TsCompiler', () => {
   describe('_makeTransformers', () => {
     test('should return the transformers object which contains before, after and afterDeclarations transformers', () => {
       const compiler = makeCompiler({
-        tsJestConfig: { ...baseTsJestConfig, isolatedModules: true, useESM: false },
+        tsJestConfig: {
+          ...baseTsJestConfig,
+          useESM: false,
+          tsconfig: {
+            isolatedModules: true,
+          },
+        },
       })
       const transformerStub = join(mockFolder, 'dummy-transformer.js')
       console.log = jest.fn()
@@ -395,7 +420,12 @@ describe('TsCompiler', () => {
     const fileName = join(mockFolder, 'thing.ts')
     const fileContent = 'const bar = 1'
     const compiler = makeCompiler({
-      tsJestConfig: { ...baseTsJestConfig, isolatedModules: true },
+      tsJestConfig: {
+        ...baseTsJestConfig,
+        tsconfig: {
+          isolatedModules: true,
+        },
+      },
     })
     const fileContentCache = new Map<string, string>()
     const fileVersionCache = new Map<string, number>()

--- a/src/legacy/compiler/ts-compiler.ts
+++ b/src/legacy/compiler/ts-compiler.ts
@@ -153,6 +153,10 @@ export class TsCompiler implements TsCompilerInstance {
         ...compilerOptions,
         module: this._ts.ModuleKind.CommonJS,
         moduleResolution,
+        /**
+         * This option is only supported in `Node16`/`NodeNext` and `Bundler` module, see https://www.typescriptlang.org/tsconfig/#customConditions
+         */
+        customConditions: undefined,
       }
     }
 
@@ -168,6 +172,10 @@ export class TsCompiler implements TsCompilerInstance {
       module: moduleKind,
       esModuleInterop,
       moduleResolution,
+      /**
+       * This option is only supported in `Node16`/`NodeNext` and `Bundler` module, see https://www.typescriptlang.org/tsconfig/#customConditions
+       */
+      customConditions: undefined,
     }
   }
 

